engine/apps/user_management/models/user.py

```diff
@@ -195,9 +195,12 @@ def google_oauth2_token_is_missing_scopes(self) -> bool:
             return False
         return not google_utils.user_granted_all_required_scopes(self.google_oauth2_user.oauth_scope)
 
-    @property
-    def avatar_full_url(self):
-        return urljoin(self.organization.grafana_url, self.avatar_url)
+    def avatar_full_url(self, organization: "Organization"):
+        """
+        Use arg `organization` instead of `self.organization` to avoid multiple requests to db when getting avatar for
+        users list
+        """
```

Comment:
May not be possible or simple, but ensuring we `select_related`  the organization for users instead should also work, right?

Reply:
It was my first thought 🙂 But we get avatar link in many places (e.g. log.author, alert_group.resolved_by_user, swap_request.beneficiary, etc.). I think: 1. it will make db queries more complex, when it's not necessary (organization is the same for every user and we already have it in memory) 2. it will be easy to forget to add `select_related` in every new place where we will need user info